rfcs/xxxx-virtual-attribute-type.md

```diff
@@ -0,0 +1,101 @@
+- Start Date: 2020-08-16
+- RFC PR: (leave this empty)
+
+# Summary
+
+It would be very useful to have "read only" attributes that are generated during "read" lifecycles (find, findOne)
+using developer-defined functions. Currently it is possible to do so for the API (by extending the result in the
+lifecycle functions). However, those attributes are not be available in any plugins (most importantly Content Manager
+and Content Type Builder), since they are not defined in the model. By adding a "virtual" attribute type, you could
+add attributes to a model without them being saved to the database, which would make them available to plugins too.
+
+Related issues: [#3131](https://github.com/strapi/strapi/issues/3131)
+
+Related PRs: [#2712](https://github.com/strapi/strapi/pull/2712), [#6433](https://github.com/strapi/strapi/pull/6433), 
+[#7221](https://github.com/strapi/strapi/pull/7221)
+
+Related discussions: [#7173](https://github.com/strapi/strapi/discussions/7173),
+[#7465](https://github.com/strapi/strapi/discussions/7465)
+
+# Example
+
+## model.settings.json
+
+```json
+{
+  ...
+  "attributes": {
+    "first_name": {
+      "type": "string",
+    },
+    "last_name": {
+      "type": "string",
+    },
+    "full_name": {
+      "type": "virtual",
+    }
+  }
+}
+```
+
+## model.js
+
+```js
+module.exports = {
+  virtualAttributes: {
+    full_name(entity) {
```

Comment:
I think here we should have getter and setter. This would allow updating database field based on a virtual field.

Reply:
That's a nice addition, though it doesn't have a lot of use cases (at least not that I can think of). How would it be best to implement in the `model.js` file though? Perhaps have two additional objects (`getters` and `setters`) in the `virtualAttributes` object, or modify the name of the virtual attribute function (`full_name_get` and `full_name_set`)?